### ISSUE TITLE
PackageName include base name

### ISSUE THREAD

**Issue by tdejager** [astral-sh/uv#2006]
Hi! We are currently integrating uv into pixi: https://github.com/prefix-dev/pixi/pull/863. Most of it works great, other than a few some things that could be improved. In RIP we also kept the source string as a base to refer back to, this is useful if people do not use a normalized name, and you want to work with the original name in some way.

See here:
https://github.com/prefix-dev/rip/blob/3d3590aac9722e219493f63f6c1fc5e9c5a91d7b/crates/rattler_installs_packages/src/types/package_name.rs

We would need something like this, we could keep the translation in our own source, but I could also add it to uv `PackageName`. Would you guys be interested in this? The trade-off is a bit more storage space for these things.

**Comment by tdejager** [comment]
Btw this is our integration branch:

https://github.com/prefix-dev/pixi/pull/863

**Comment by konstin** [comment]
The main reason why we eventually chose not to include it is that different source can disagree about what the real name is, e.g. pypi says something different than the wheel metadata than a different index. Do you have a reliable source of display names? I'm also thinking about how we'd sync package name instances from different source.

**Comment by tdejager** [comment]
Yeah so we didn't really worry about it, in the sense that we just wanted to know what the *original* name was. This is useful for keeping track of how it was specified in a manifest format.

I think the problem you are describing is a different one. If you don't feel like it makes sense within the uv context, then we can wrap it in a wrapper struct on our end :)

**Comment by baszalmstra** [comment]
Similar to `VerbatimUrl` basically. We want to store the normalized package name as well as the original string it was created from.
